### PR TITLE
Create scorecards-analysis.yml

### PR DESCRIPTION
**Pull request by godofredoc**
Enables scorecards code scanning for the flutter repository. This will allow us to get automated security reports for flutter/flutter using "OSSF Scorecards supply-chain security analysis"

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Hixie** [comment]
Can we enable this for the entire org instead of one repo, e.g. [the way we do with `.allstar`](https://github.com/flutter/.allstar)?

**Comment by godofredoc** [comment]
> Can we enable this for the entire org instead of one repo, e.g. [the way we do with `.allstar`](https://github.com/flutter/.allstar)?

Unfortunately github action workflows are not supported at the org level.

File: .github/workflows/scorecards-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  analysis:
+    name: Scorecards analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
```
Comment (Hixie) [inline .github/workflows/scorecards-analysis.yml:19]: i couldn't figure out what this actually enabled

File: .github/workflows/scorecards-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@c8416b0b2bf627c349ca92fc8e3de51a64b005cf # v1.0.2
+        with:
+          results_file: results.sarif
+          results_format: sarif
+          # Read-only PAT token. To create it,
+          # follow the steps in https://github.com/ossf/scorecard-action#pat-token-creation.
+          repo_token: ${{ secrets.SCORECARD_READ_TOKEN }}
```
Comment (Hixie) [inline .github/workflows/scorecards-analysis.yml:36]: does this have a lifetime?

**Review by Hixie — APPROVED** [review]
LGTM modulo that I don't know what `security-events: write` means

File: .github/workflows/scorecards-analysis.yml (hunk 1 above)
Comment (godofredoc) [reply .github/workflows/scorecards-analysis.yml:19]: It grants access to the following APIs: https://docs.github.com/en/rest/reference/permissions-required-for-github-apps#permission-on-security-events

File: .github/workflows/scorecards-analysis.yml (hunk 2 above)
Comment (godofredoc) [reply .github/workflows/scorecards-analysis.yml:36]: We documented this to be 6 months TTL

**Comment by godofredoc** [comment]
> LGTM modulo that I don't know what `security-events: write` means

Added a link to the documentation. It grants the actions workflow write access to the github security API

**Pull request merged**